### ISSUE TITLE
Better solution for crypto lib

### ISSUE THREAD

**Issue by boundsj** [boundsj/ObjectiveDDP#46]
Currently, for auth with meteor servers, ObjectiveDDP uses a compiled for iOS (by me) libcrypto binary and openssl headers. Also, it uses an srp library (customized for meteor by me).

A new solution should be created with the following points in mind:
1. There is nothing in the ObjectiveDDP documentation that warns inexperienced users that they might want to be wary of including precompiled crypto binaries in their application (I may or may not be trustworthy and they may want to compile and include their own libraries and headers)
2. Related to issue #1 the auth/crypto binaries and headers come part and parcel with ObjectiveDDP. It would be nice to separate them - there are several OpenSSL pods available (but users should still be conscious of the security implications of that as noted above)
3. Although it might be slightly avantgarde, an alternative solution could be to actually reuse Meteor auth/SRP logic (from their JavaScript libraries) with JavaScriptCore - this could be a separate project/pod that would focus specifically on meteor SRP and making meteor core auth functions in javascript available to ObjC clients. Users could optionally include this style pod (instead of a C / OpenSSL one). Of course, there are security issues with this approach, too.

**In short, ObjectiveDDP should not embed the auth code - it should function without it for non auth use cases and the docs should suggest one or more pods for auth and give instructions for users to compile and include their own auth package.**

NOTE: As of November 2013, since we do currently have a working auth solution (even though it is embedded) I plan to deprioritize this for a while longer while we work on the main API.
